lib/shared/drive-constraints.js

```diff
@@ -232,3 +232,153 @@ exports.isDriveValid = (drive, image) => {
 exports.isDriveSizeRecommended = (drive, image) => {
   return _.get(drive, [ 'size' ], UNKNOWN_SIZE) >= _.get(image, [ 'recommendedDriveSize' ], UNKNOWN_SIZE);
 };
+
+/**
+ * @summary Drive/image compatibility status messages.
+ * @public
+ * @type {Object}
+ *
+ * @description
+ * Status messages intended to be shown to the user.
+ */
+exports.COMPATIBILITY_STATUS_MESSAGES = {
+
+  /**
+   * @property {String} SIZE_NOT_RECOMMENDED
+   * @memberof COMPATIBILITY_STATUS_MESSAGES
+   *
+   * @description
+   * The image and drive compatibility is not recommended; happens when the
+   * actual drive size is smaller than the image's recommended drive size.
+   */
+  SIZE_NOT_RECOMMENDED: 'Not Recommended',
+
+  /**
+   * @property {String} TOO_SMALL
+   * @memberof COMPATIBILITY_STATUS_MESSAGES
+   *
+   * @description
+   * The drive is too small for the image.
+   */
+  TOO_SMALL: 'Too Small For Image',
+
+  /**
+   * @property {String} LOCKED
+   * @memberof COMPATIBILITY_STATUS_MESSAGES
+   *
+   * @description
+   * The drive is locked (e.g. the lock-tab on SD cards) and cannot be written to.
+   */
+  LOCKED: 'Locked',
+
+  /**
+   * @property {String} SYSTEM
+   * @memberof COMPATIBILITY_STATUS_MESSAGES
+   *
+   * @description
+   * The drive is a system drive and should not be written to.
+   */
+  SYSTEM: 'System Drive',
+
+  /**
+   * @property {String} CONTAINS_IMAGE
+   * @memberof COMPATIBILITY_STATUS_MESSAGES
+   *
+   * @description
+   * The drive contains the image and therefore cannot be written to.
+   */
+  CONTAINS_IMAGE: 'Drive Contains Image'
+};
+
+/**
+ * @summary Drive/image compatibility status types.
+ * @public
+ * @type {Object}
+ *
+ * @description
+ * Status types classifying what kind of message it is, i.e. ok, error, warning.
+ */
+exports.COMPATIBILITY_STATUS_TYPES = {
+  OK: 0,
+  WARNING: 1,
+  ERROR: 2
+};
+
+/**
+ * @summary Get drive/image compatibility in an object
+ * @function
+ * @public
+ *
+ * @description
+ * Given an image and a drive, return their compatibility status object
+ * containing the status type (ERROR, WARNING, and OK), and accompanying
+ * status message.
+ *
+ * @param {Object} drive - drive
+ * @param {Object} image - image
+ * @returns {Object[]} list of compatibility status objects
+ *
+ * @example
+ * const drive = {
+ *   device: '/dev/disk2',
+ *   name: 'My Drive',
+ *   size: 4000000000
+ * };
+ *
+ * const image = {
+ *   path: '/path/to/rpi.img',
+ *   size: 2000000000
+ *   recommendedDriveSize: 4000000000
+ * });
+ *
+ * const statuses = constraints.getDriveImageCompatibilityStatuses(drive, image);
+ *
+ * for ({ type, message } of statuses) {
+ *   if (type === constraints.COMPATIBILITY_STATUS_TYPES.WARNING) {
+ *     // do something
+ *   } else if (type === constraints.COMPATIBILITY_STATUS_TYPES.ERROR) {
+ *     // do something else
+ *   }
+ * }
+ */
+exports.getDriveImageCompatibilityStatuses = (drive, image) => {
```

Comment:
I tried this, but the line ended up longer than before when using `.getDriveImageCompatibilityStatuses()` in `.isDriveValid()` and it'd be slightly less efficient because we'd also check for warnings. Maybe you had something else in mind though!